src/azure-cli/azure/cli/command_modules/appservice/access_restrictions.py

```diff
@@ -0,0 +1,163 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from knack.util import CLIError
+from azure.cli.core.commands import LongRunningOperation
+from azure.mgmt.web.models import IpSecurityRestriction
+from azure.mgmt.network.models import ServiceEndpointPropertiesFormat
+from azure.cli.command_modules.appservice.custom import get_site_configs
+from azure.cli.command_modules.appservice._appservice_utils import _generic_site_operation
+from azure.cli.command_modules.network._client_factory import network_client_factory
+
+logger = get_logger(__name__)
+
+
+def show_webapp_access_restrictions(cmd, resource_group_name, name, slot=None):
+    import json
+    configs = get_site_configs(cmd, resource_group_name, name, slot)
+    access_restrictions = json.dumps(configs.ip_security_restrictions, default=lambda x: x.__dict__)
+    scm_access_restrictions = json.dumps(configs.scm_ip_security_restrictions, default=lambda x: x.__dict__)
+    access_rules = {
+        "scmIpSecurityRestrictionsUseMain": configs.scm_ip_security_restrictions_use_main,
+        "ipSecurityRestrictions": json.loads(access_restrictions),
+        "scmIpSecurityRestrictions": json.loads(scm_access_restrictions)
+    }
+    return access_rules
+
+
+def add_webapp_access_restriction(
+        cmd, resource_group_name, name, rule_name, priority,
+        action='Allow', ip_address=None, subnet=None,
+        vnet_name=None, description=None, scm_site=False,
+        ignore_missing_vnet_service_endpoint=False, slot=None):
+    configs = get_site_configs(cmd, resource_group_name, name, slot)
+
+    # get rules list
+    access_rules = configs.scm_ip_security_restrictions if scm_site else configs.ip_security_restrictions
+    # check for null
+    access_rules = access_rules if access_rules else []
```

Comment:
nit:  `access_rules = access_rules or []`

Reply:
Learning new language constructs in Python - thanks